### ISSUE TITLE
Global header: form elements must have labels

### ISSUE THREAD

**Issue by jcjuett** [salesforce/design-system-react#1844]
http://design-system-react-components.herokuapp.com/?selectedKind=SLDSGlobalHeader&selectedStory=Search%20%2B%20Navigation&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

HTML:  <input aria-activedescendant="" aria-autocomplete="list" aria-expanded="false" class="slds-input" id="undefinedLookup" placeholder="Search Salesforce" role="combobox" type="text" value="">

Success Criteria: Screen reader reads descriptive label for form field. The label elements for attribute matches the id attribute of the input element.

Search doesn't have a `label` node (it can be invisible).

**Comment by sharaththegeek** [comment]
Why does the form field in this scenario need a separate label when the placeholder successfully describes the intent of the field?

**Comment by interactivellama** [comment]
@sharaththegeek Thanks for asking!

This is an accessibility issue for keyboard users using assistive technology such as screen readers.

> Placeholder text disappears when a text box receives focus:
> For keyboard users this can be problematic as they must read ahead of current focus when filling in forms.
> Users, especially those with memory impairments will not have the text label available for reference at the same time as filling in a field.

https://developer.paciellogroup.com/blog/2011/02/html5-accessibility-chops-the-placeholder-attribute/

**Comment by sharaththegeek** [comment]
Oh I understand now! 
Thank you so much!

**Comment by interactivellama** [comment]
@kevinparkerson I'm stealing this one back in order to make aXe testing pass on all examples.
